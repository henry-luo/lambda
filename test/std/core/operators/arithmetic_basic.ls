// Test: Basic Arithmetic Operations
// Category: core/operators
// Type: positive
// Expected: 32

// Test basic arithmetic operations
let a = 10
a
let b = 5
b
let result = (a + b) * 2 + 2
result
result
