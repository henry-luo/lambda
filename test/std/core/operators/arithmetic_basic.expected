09:15:47 [NOTICE] Start transpiling test/std/core/operators/arithmetic_basic.ls...
[script:list]
  [content:list[4]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:result:int]
        [binary expr +.3:int]
          [binary expr *.5:int]
            [primary expr:int,const:0]
              [binary expr +.3:int]
                [primary expr:int,const:1]
                  [ident:a:int,const:1]
                [primary expr:int,const:1]
                  [ident:b:int,const:1]
            [primary expr:int,const:1]
              [const: int]
          [primary expr:int,const:1]
            [const: int]
    item:
    [primary expr:int,const:0]
      [ident:result:int,const:0]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/arithmetic_basic.ls' to MIR
C2MIR init end           -- 68 usec
test/std/core/operators/arithmetic_basic.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2429, parse tokens -- 1459
  C2MIR preprocessor end    -- 565 usec
  C2MIR parser end          -- 688 usec
  C2MIR context checker end -- 776 usec
  C2MIR generator end       -- 833 usec
C2MIR compiler end                -- 867 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/arithmetic_basic.ls' executed: #####
32
